gateway/plugin.go

```diff
@@ -0,0 +1,53 @@
+package gateway
+
+import (
+	"context"
+
+	gql "github.com/machinebox/graphql"
+	"github.com/railwayapp/cli/entity"
+	"github.com/railwayapp/cli/errors"
+)
+
+func (g *Gateway) GetAvailablePlugins(ctx context.Context, projectId string) ([]*string, error) {
+	gqlReq := gql.NewRequest(`
+		query ($projectId: ID!) {
+			availablePluginsForProject(projectId: $projectId)
+		}
+	`)
+
+	gqlReq.Var("projectId", projectId)
+
+	g.authorize(ctx, gqlReq.Header)
+
+	var resp struct {
+		Plugins []*string `json:"availablePluginsForProject"`
```

Comment:
What's the utility of having an array of pointers here vs just an array of strings?

Reply:
was just trying to follow convention. good point though; ran into some memory issues along the way.

Reply:
Yea. You'll also just be derefing things for no reason (like in your plugin search function)